gui/slick/views/schedule.mako

```diff
@@ -233,11 +233,13 @@
             % endif
         % endif
     % elif 'date' == sort:
-        <% cur_ep_airdate = cur_result['localtime'].date() %>
+        <%
+            cur_ep_airdate = cur_result['localtime'].date()
+            cur_ep_enddate = cur_result['localtime'] + datetime.timedelta(minutes = run_time)
+        %>
 
         % if cur_segment != cur_ep_airdate:
             %if run_time:
```

Comment:
@fernandog Otherwise it would only set the enddate for the first episode in the segment (ie Today, Missed, ...), and then further down the code it would use the enddate of the first episode of that segment for all episodes in that segment. Thus, when one episode finished airing, all others for that day would get marked in red as well, even when they weren't aired yet, because their enddate would not get set correctly (because that part would get skipped because the day header was already calculated) Only happened on banner/poster view and when sorted on date, other view modes weren't affected.
However I now notice this introduces some code redundancy so let me factor that out first...